src/Engine/ProtoCore/Lang/CallSite.cs

```diff
@@ -1853,6 +1869,58 @@ private FunctionGroup FirstFunctionGroupInInheritanceChain(RuntimeCore runtimeCo
             return ret;
         }
 
+        /// <summary>
+        /// Determine if the formalParameters are homogeneous and initialize a flat list of final formalParameters
+        /// </summary>
+        /// <returns>true if the formalParameters are homogeneous</returns>
+        private static bool AreParametersHomogeneous(List<StackValue> formalParameters, RuntimeCore runtimeCore, List<StackValue> finalFormalParameters)
```

Comment:
feels weird to me that this has side effects on a list you need to initialize and pass in to this method... what about just returning a tuple?